src/System.Console/src/System/IO/StdInStreamReader.cs

```diff
@@ -0,0 +1,259 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.Win32.SafeHandles;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Runtime.InteropServices;
+
+namespace System.IO
+{
+    /* This class is used by for reading from the stdin.
+     * It is designed to read stdin in raw mode for interpreting
+     * key press events and maintain its own buffer for the same.
+     * which is then used for all the Read operations
+     */
+    internal class StdInStreamReader : StreamReader
+    {
+        private char[] _unprocessedBufferToBeRead; // Buffer that might have already been read from stdin but not yet processed.
+        private const int BytesToBeRead = 255; // No. of bytes to be read from the stream at a time.
+        private int _startIndex; // First unprocessed index in the buffer;
+        private int _endIndex; // Last unprocessed index in the buffer;
+        private StringBuilder _readLineSB;
+
+        internal StdInStreamReader(Stream stream, Encoding encoding, int bufferSize) : base(stream: stream, encoding: encoding, detectEncodingFromByteOrderMarks: false, bufferSize: bufferSize, leaveOpen: true)
+        {
+            _unprocessedBufferToBeRead = new char[encoding.GetMaxCharCount(bufferSize)];
```

Comment:
Should this be encoding.GetMaxCharCount(BytesToBeRead) rather than encoding.GetMaxCharCount(bufferSize)?